app/code/Magento/NewsletterApi/Exception/AlreadySubscribedException.php

```diff
@@ -0,0 +1,14 @@
+<?php
+/**
```

Comment:
> Business logic (both application and domain) MUST NOT be managed with exceptions. Conditional statements should be used instead.
It's funny you mention that. Please take a look at the [following piece of code](https://github.com/magento/magento2/blob/99e85cbc45223baa3551e4c534b650c0d2c6358b/app/code/Magento/Quote/Model/Webapi/ParamOverriderCartId.php#L58-L60)
Could you guess _which kind_ of entity wasn't found? Hint: it's not necessarily the Customer, it might be Product as well.

Would you say this code is badly written and should avoid potentially provoking an exception as means to implement conditional logic? There's [a lot of similar code](https://github.com/magento/magento2/search?q=catch+NoSuchEntityException+&type=Code&utf8=%E2%9C%93) all around. Is there a sane alternative to prefer in such cases?